### PR TITLE
Implement Custom Properties

### PR DESCRIPTION
**Pull request by liaodaniel**
Resolves https://github.com/google/go-github/issues/2965

### CODE REVIEW

File: github/orgs_properties.go
Comment (gmlewis) [inline github/orgs_properties.go:15]: This field is required, so let's please make it mandatory.

```suggestion
	PropertyName string `json:"property_name"`
```

File: github/orgs_properties.go
Comment (gmlewis) [inline github/orgs_properties.go:35]: Looking at the [official documentation](https://docs.github.com/en/rest/orgs/custom-properties?apiVersion=2022-11-28#list-custom-property-values-for-organization-repositories), it looks like `value` can be `null`, so let's please make this field nullable.

The "Response schema" tab says this:

```json
      "properties": {
        "type": "array",
        "items": {
          "title": "Custom Property Value",
          "description": "Custom property name and associated value",
          "type": "object",
          "properties": {
            "property_name": {
              "type": "string",
              "description": "The name of the property"
            },
            "value": {
              "type": [
                "string",
                "null"
              ],
              "description": "The value assigned to the property"
            }
          },
          "required": [
            "property_name",
            "value"
          ]
        },
```

```suggestion
	Value        *string `json:"value,omitempty"`
```

File: github/orgs_properties.go
Comment (gmlewis) [inline github/orgs_properties.go:38]:
```suggestion
// GetAllCustomProperties gets all custom properties that are defined for the specified organization.
```

File: github/orgs_properties.go
Comment (gmlewis) [inline github/orgs_properties.go:115]: Can we please change `name` here to `customPropertyName` just to make it super-clear?
I only ask because it is so common in this repo to have something like `org, repo string` that I know I would personally probably fill in the repo name here by mistake unless I see that glaringly obvious parameter name.

```suggestion
func (s *OrganizationsService) CreateOrUpdateCustomProperty(ctx context.Context, org, customPropertyName string, property *CustomProperty) (*CustomProperty, *Response, error) {
```

File: github/orgs_properties.go
Comment (gmlewis) [inline github/orgs_properties.go:137]: Same here, please?

```suggestion
func (s *OrganizationsService) RemoveCustomProperty(ctx context.Context, org, customPropertyName string) (*Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @liaodaniel!
Just a few minor tweaks, please, and then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

File: github/orgs_properties.go
Comment (liaodaniel) [reply github/orgs_properties.go:15]: This was a conscious change due to https://docs.github.com/en/rest/orgs/custom-properties?apiVersion=2022-11-28#create-or-update-a-custom-property-for-an-organization. This API relies on `property_name` to be specified in the path instead of the field to be in the custom property object.

Comment (liaodaniel) [reply github/orgs_properties.go:15]: Hmm, just to clarify. Would it be fine to update it to send the unused field `property_name` as part of the body?

Comment (gmlewis) [reply github/orgs_properties.go:15]: Ah, I see what you are talking about.

How about just add a comment to this field... something like:
```go
// Although PropertyName is required for most endpoints, this struct is shared with the PUT endpoint
// where it is sent in the path, and is therefore a pointer with "omitempty".
```

Comment (liaodaniel) [reply github/orgs_properties.go:17]: Thank you @gmlewis. I'll update it with something along those lines.

**Review by gmlewis — APPROVED** [review]
Thank you, @liaodaniel !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

Also, can you please switch this PR out of "Draft" mode when you are ready?

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
